R/betydb.R

```diff
@@ -159,11 +160,113 @@ betydb_search <- function(query = "Maple SLA", ..., include_unchecked = NULL){
 }
 
 betydb_GET <- function(url, args = list(), key = NULL, user = NULL, pwd = NULL, which, ...){
-  txt <- betydb_http(url, args, key, user, pwd, ...)
-  lst <- jsonlite::fromJSON(txt, simplifyVector = TRUE, flatten = TRUE)
+
+  api_version <- ifelse(grepl('/beta/api', url), 'beta', 'v0')
+  if(!exists('per_call_limit')) {
+    per_call_limit <- 5000
+  }
+
+  if(is.null(api_version)){
+    api_version <- options()$betydb_api_version
+  }
+  if(api_version == 'v0'){
+    txt <- betydb_http(url, args, key, user, pwd, ...)
+    lst <- jsonlite::fromJSON(txt, simplifyVector = TRUE, flatten = TRUE)
+  } else if (api_version == 'beta'){
+
+    if(is.null(args$limit)) {
+      args$limit <- 200
+    } else if (args$limit == 'none'){
+      args$limit <- 10^9
+    } else if(!is.na(as.numeric(args$limit))){
+      args$limit <- as.numeric(args$limit)
+    } else {
+      stop('invalid value given for limit', ifelse(is.null(args$limit), "NULL", args$limit),
+           "\nlimit must be a positive integer or 'none'")
+    }
+
+    if(args$limit <= per_call_limit){
+      txt <- betydb_http(url, args, key, user, pwd, ...)
+      lst <- jsonlite::fromJSON(txt, simplifyVector = TRUE, flatten = TRUE)
+    } else if (args$limit > per_call_limit){ # divide large requests (aka page)
```

Comment:
This approach makes it more clear and explicit the case under which we begin paging. the final else on line 264 below captures if `args$limit` captures cases where `!args$limit <= per_call_limit | args$limit > per_call limit`. (e.g. if `args$limit` can't be coerced to a numeric value)